### PR TITLE
Reland "Output .js files as ES6 modules. (#52023)"

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
This is an attempt to reland https://github.com/flutter/engine/pull/52023. The issue previously is that if it was not specified by the user, the default CanvasKit base URL did not have a leading slash, which does not work when doing dynamic imports.

### CODE REVIEW

File: lib/web_ui/lib/src/engine/configuration.dart [hunk 1]
```diff
@@ -241,7 +241,7 @@ class FlutterConfiguration {
       _configuration?.canvasKitBaseUrl ?? _defaultCanvasKitBaseUrl;
   static const String _defaultCanvasKitBaseUrl = String.fromEnvironment(
     'FLUTTER_WEB_CANVASKIT_URL',
-    defaultValue: 'canvaskit/',
+    defaultValue: '/canvaskit/',
```
Comment (ditman) [inline lib/web_ui/lib/src/engine/configuration.dart:244]: I think this'd break the base-url functionality.

Should this be `/$BASE_URL/canvaskit/` ?

Comment (eyebrowsoffire) [reply lib/web_ui/lib/src/engine/configuration.dart:244]: We discussed this offline, I think this current behavior is a reasonable default and is consistent with what flutter.js is doing. In the future, we may want to deal with the base URI stuff in a more robust way. Our current behavior is not super consistent in general.

**Review by ditman — APPROVED** [review]
Ship it!

**Comment by jiahaog** [comment]
Reason for revert: canvaskit.js cannot be loaded in an internal end to end test - see b/350885206

**Pull request merged**